### PR TITLE
nvidia-gpu 2.0 make compatible for k8s 

### PR DESCRIPTION
**Pull request by CodeJuan**
### Ⅰ. Describe what this PR did
[k8s-device-plugin](https://github.com/NVIDIA/k8s-device-plugin) uses environment variables to specify a GPU accelerated container, so we should support it.
In order to be compatible with k8s, pouchd should set nvidia prestart hook if nvidia environment variable was set by user.

### Ⅳ. Describe how to verify it
1. Prerequisites:
    - An instance with a NVIDIA GPU
    - The appropriate GPU driver is installed
2. Run GPU container with nvidia env.
```
pouch run -it -e NVIDIA_VISIBLE_DEVICES=all -e NVIDIA_DRIVER_CAPABILITIES=all centos:7 bash
```
3. Exec nvidia-smi in container.
4. If nvidia-smi return error `NVIDIA-SMI couldn't find libnvidia-ml.so library in your system. Please make sure that the NVIDIA Display Driver is properly installed and present in your system`, then add a link to libnvidia-ml.so
```
version=`nvidia-smi --help | head -n 1 | awk -F "-- v" '{print $2}'`
ln -s  /usr/lib64/libnvidia-cfg.so.1 /usr/lib64/libnvidia-cfg.so
ln -s /usr/lib64/libnvidia-cfg.so."$version" /usr/lib64/libnvidia-cfg.so.1
ln -s  /usr/lib64/libnvidia-ml.so.1 /usr/lib64/libnvidia-ml.so
ln -s /usr/lib64/libnvidia-ml.so."$version" /usr/lib64/libnvidia-ml.so.1
```

### CODE REVIEW

File: daemon/mgr/spec_nvidia_hook_test.go [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+	"github.com/opencontainers/runtime-spec/specs-go"
+)
+
+func Test_setNvidiaHook(t *testing.T) {
+	nvidiaHookName = "test-nvidia-container-runtime-hook"
+	installDir := "/usr/local/bin/"
+	fullname := path.Join(installDir, nvidiaHookName)
+	os.Remove(fullname)
+	os.Create(fullname)
+	os.Chmod(fullname, 0755)
```
Comment (HusterWan) [inline daemon/mgr/spec_nvidia_hook_test.go]: How about adding `defer os.Remove(fullname)` here?  delete the `test-nvidia-container-runtime-hook` after test finished

Comment (CodeJuan) [reply daemon/mgr/spec_nvidia_hook_test.go]: Actually, there is defer function at line22. 
Sometimes test was broken unexpected, we should delete the mock file at first.

File: daemon/mgr/spec_nvidia_hook.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import (
+	"os/exec"
+
+	"github.com/alibaba/pouch/pkg/utils"
+
+	"github.com/opencontainers/runtime-spec/specs-go"
+)
+
+var (
+	nvidiaHookName = "nvidia-container-runtime-hook"
```
Comment (allencloud) [inline daemon/mgr/spec_nvidia_hook.go]: please add more description to this variable. Is this is local file when executing hook?

File: docs/features/pouch_with_gpu.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
+# PouchContainer with GPU
+
```
Comment (allencloud) [inline docs/features/pouch_with_gpu.md]: Before the graph, I think we should add a general introduction of the usage of PouchContainer with GPU.

File: daemon/mgr/spec_nvidia_hook.go (hunk 2 above)
Comment (CodeJuan) [reply daemon/mgr/spec_nvidia_hook.go]: Both nvidia-container-runtime-hook and nvidia-container-cli was packaged in pouch rpm.

File: docs/features/pouch_with_gpu.md (hunk 3 above)
Comment (CodeJuan) [reply docs/features/pouch_with_gpu.md]: done

File: daemon/mgr/spec_nvidia_hook_test.go (hunk 1 above)
Comment (CodeJuan) [reply daemon/mgr/spec_nvidia_hook_test.go]: Take it easy man 😁

**Comment by zhuangqh** [comment]
UT fails @CodeJuan 
```
=== RUN   Test_setNvidiaHook
--- FAIL: Test_setNvidiaHook (0.00s)
	spec_nvidia_hook_test.go:108: setNvidiaHook = exec: "test-nvidia-container-runtime-hook": executable file not found in $PATH, want <nil>
	spec_nvidia_hook_test.go:111: setNvidiaHook = [], want [{ [ prestart] [] <nil>}]
	spec_nvidia_hook_test.go:108: setNvidiaHook = exec: "test-nvidia-container-runtime-hook": executable file not found in $PATH, want <nil>
	spec_nvidia_hook_test.go:111: setNvidiaHook = [], want [{ [ prestart] [] <nil>}]
FAIL
coverage: 10.8% of statements
FAIL	github.com/alibaba/pouch/daemon/mgr	0.593s
make: *** [unit-test] Error 1
```

**Comment by zhuangqh** [comment]
LGTM

**Pull request merged**